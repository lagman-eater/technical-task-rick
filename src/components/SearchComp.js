import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { setSearch } from '../redux/searchSlicer'
import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import { setPage } from '../redux/pageSlicer';

const PaginationComp = () => {
    const dispatch = useDispatch()

    const search = useSelector(state => state.search)
    const handleSearch = ((e, value) => {
        dispatch(setSearch(e.target.value))
        dispatch(setPage(1))
    })

    return (
        <Box
            component="form"
            sx={{ '& > :not(style)': { m: 1, width: '25ch' } }}
            noValidate
            autoComplete="off"
        >
            <TextField
                id="outlined-basic"
                label="Search by name"
                variant="outlined"
                value={search.search !== '' ? search.search : ''}
                onChange={(e) => handleSearch(e)}
            />
        </Box>
    )
}

export default PaginationComp