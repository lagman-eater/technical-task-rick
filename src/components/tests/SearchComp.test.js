import { render } from "@testing-library/react";
import * as reduxHooks from 'react-redux'
import SearchComp from "../SearchComp";

jest.mock('react-redux')

describe('SearchComp', () => {
    it('Should make search possible for empty string', () => {
        jest.spyOn(reduxHooks, 'useSelector').mockReturnValue('')
        const component = render(<SearchComp />)
        expect(component).toMatchSnapshot()
    })
    it('Should make the search possible for some string', () => {
        jest.spyOn(reduxHooks, 'useSelector').mockReturnValue('Rick')
        const component = render(<SearchComp />)
        expect(component).toMatchSnapshot()
    })
})
