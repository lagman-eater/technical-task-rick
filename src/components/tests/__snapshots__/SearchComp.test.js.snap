// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComp Should make search possible for empty string 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        autocomplete="off"
        class="MuiBox-root css-xn6ruc"
        novalidate=""
      >
        <div
          class="MuiFormControl-root  MuiTextField-root  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined  MuiFormLabel-colorPrimary MuiFormLabel-filled  MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined  css-18pjc51-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            Search by name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root  MuiInputBase-colorPrimary MuiInputBase-formControl  css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input   css-1jk99ih-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline  css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-kg5swy"
              >
                <span>
                  Search by name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      autocomplete="off"
      class="MuiBox-root css-xn6ruc"
      novalidate=""
    >
      <div
        class="MuiFormControl-root  MuiTextField-root  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined  MuiFormLabel-colorPrimary MuiFormLabel-filled  MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined  css-18pjc51-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="outlined-basic"
          id="outlined-basic-label"
        >
          Search by name
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root  MuiInputBase-colorPrimary MuiInputBase-formControl  css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input   css-1jk99ih-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-basic"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline  css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-kg5swy"
            >
              <span>
                Search by name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SearchComp Should make the search possible for some string 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        autocomplete="off"
        class="MuiBox-root css-xn6ruc"
        novalidate=""
      >
        <div
          class="MuiFormControl-root  MuiTextField-root  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined  MuiFormLabel-colorPrimary MuiFormLabel-filled  MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined  css-18pjc51-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            Search by name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root  MuiInputBase-colorPrimary MuiInputBase-formControl  css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input   css-1jk99ih-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline  css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-kg5swy"
              >
                <span>
                  Search by name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      autocomplete="off"
      class="MuiBox-root css-xn6ruc"
      novalidate=""
    >
      <div
        class="MuiFormControl-root  MuiTextField-root  css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined  MuiFormLabel-colorPrimary MuiFormLabel-filled  MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined  css-18pjc51-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="outlined-basic"
          id="outlined-basic-label"
        >
          Search by name
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root  MuiInputBase-colorPrimary MuiInputBase-formControl  css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input   css-1jk99ih-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-basic"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline  css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-kg5swy"
            >
              <span>
                Search by name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
